### ISSUE TITLE
508-defect-3 [COGNITION]: Edit, cancel, and remove buttons SHOULD be styled as buttons

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19562]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## Details

The "edit" buttons on this page are styled as links which is materially dishonest causing several issues:
- sighted users may think the "editing" action will occur on a new page
- voice interface users may not be able to activate the button should they say "click on edit link"
- screen reader users may be confused on help calls if they are instructed to click on an edit link when the screen reader announces it as a button
- low vision screen reader users may be confused that the screen reader is announcing a visually styled link as a button
- breaks consistency with other edit buttons across the interface

"Cancel" buttons on this page may face less of the issues listed above as they may be less likely to be misinterpreted as a link (e.g. hard to picture "cancelling" something on another page vs. editing)-- however, they still may be prone to misinterpretation and are still materially dishonest.

For more on this long-tail effort, check out our [buttons vs. links Mural board](https://app.mural.co/t/vsa8243/m/vsa8243/1610052905994/253723ef4b2ee9a726f538876a257df94d468db1) and this [Slack thread discussing _why_ material honesty is so important](https://dsva.slack.com/archives/C8E985R32/p1610659601009200?thread_ts=1610652897.002600&cid=C8E985R32).

"A few years at an accessibility event, a blind man was telling a story about how he was on the phone with someone who was trying to help him sign up for a service. The person on the phone told him to click the “sign up button” and he couldn’t find it because it was a link. He told the story with a lot more gravity than I can, but that was the point where I realized how important the human-human communication part of accessibility is, it’s not just machine to human." - @timwright12 

## Acceptance Criteria

- [ ] "Edit" buttons are styled as buttons
- [ ] "Cancel" buttons are styled as buttons

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/profile/direct-deposit` in browser
2. Inspect "edit"
3. Confirm it's a button

## Solution (if known)

Style the link as a button, options may include:

- Using primary or secondary outline button styles
- Using an icon + text pairing

## WCAG or Vendor Guidance (optional)

* [Adam Silver: But sometimes buttons look like links](https://adamsilver.io/articles/but-sometimes-buttons-look-like-links/)
* [VA Design System guidance on buttons](https://design.va.gov/components/buttons#guidance)

## Screenshots or Trace Logs

<img width="1667" alt="Screen Shot 2021-02-05 at 1 10 10 PM" src="https://user-images.githubusercontent.com/14154792/107072766-72a32200-67b4-11eb-85ab-7cc2afd60ac8.png">

**Comment by mshea0606** [comment]
@callen2563 to meet with @tressaellen and discuss next steps

**Comment by andaleliz** [comment]
@tressaellen just getting caught up - did you and Cassandra talk about this at all yet?

**Comment by andaleliz** [comment]
Tressa and I talked; [trying to dig into the history of this design decision on Slack](https://dsva.slack.com/archives/C0NGDDXME/p1615236395028300) as there was some back and forth about it before landing on this solution.  Buttons were explored originally but determined to be overwhelming/undesirable as a big ol' stack of buttons.

I've seen a lot of convos around this issue on the site, so also seeking to determine whether or not VSP has guidance or is working toward some. It appears they're working on an "action link" solution.

I'll take the lead on resolving this issue for now.  Will see what I can learn about previous design decisions before moving forward.

**Comment by andaleliz** [comment]
It sounds like the design system team will have a pattern + documentation for this approach in 2ish weeks.   [From @CrystabelReiter in Slack on 3/8](https://dsva.slack.com/archives/C0NGDDXME/p1615246653040000?thread_ts=1615236523.030100&cid=C0NGDDXME):
>I'll be finished with the documentation draft tomorrow morning and then the design system team will submit for a staging review. Once staging review is complete, we'll probably need some time to incorporate collab cycle feedback, after that-- it'll be available for teams. I'm guessing this could take 2ish weeks? We're trying our best to make sure we get this to ya'll as soon as possible, but still making sure that the documentation makes sense and is good quality. 

@Samara-Strauss I don't get the impression that this issue is urgent or time-sensitive. Is it okay for this to stay on hold until later this month, when we have access to the design system update that will address this?

**Comment by Samara-Strauss** [comment]
Sounds good! I'll move it out of next sprint.

**Comment by andaleliz** [comment]
Related (possibly should be a separate ticket): cancel buttons are treated differently in different parts of the profile. E.g. in direct deposit, we've got cancel as a text link style, and in address it's styled as a button.  
![Screen Shot 2021-03-18 at 10 18 06 AM](https://user-images.githubusercontent.com/55992116/111788276-4809b600-8896-11eb-9ed2-2b1c2bb2017f.png)
![Screen Shot 2021-03-18 at 10 17 59 AM](https://user-images.githubusercontent.com/55992116/111788287-4b9d3d00-8896-11eb-9f57-5e310cd9358c.png)

We'll need to look through the screens and/or code and identify all the places this needs to be cleaned up.

**Comment by andaleliz** [comment]
@Samara-Strauss @ajakabcin The design system team has put a solution for this in their backlog with no ETA, so I think we should move forward - unless it's okay for it to sit in ours as well.

I'll have bandwidth to work on something aside from facilities and communications this sprint.  Not sure if it should be this, or the international phone number issue already in the sprint.

**Comment by andaleliz** [comment]
FYI this is the relevant design system team issue where they're tracking their work on a lighter-weight button: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/406

**Comment by Samara-Strauss** [comment]
> The design system team has put the new button design in their backlog with no ETA, so I think we should move forward - unless it's okay for it to sit in ours as well.

Sorry, I am not making the connection I need to make. How does this affect this ticket? Are we able to make progress on this without the new button design?

**Comment by andaleliz** [comment]
Yes, I can think of a couple options if we want to make progress ASAP.  We were waiting on this because we thought the design system team was going to have guidance within a couple of weeks.  

- we explore a lighter weight button design, which we can submit for consideration to be added to the design system if it tests well. I _think_ this option will require some review steps w/ the collab cycle since we'd be going "off-brand" and coming up with something custom.  We could test it when we do research for any of the upcoming profile work.
- don't spend much design time on this and replace all edit/cancel buttons styled as links to the secondary button style already in the design system. I believe this was explored in earlier iterations of the design, and the team decided against it for edit buttons because it was overwhelming on the page. I think the phrase "wall of buttons" came up when I asked about this in the past.

**Comment by Samara-Strauss** [comment]
> we explore a lighter weight button design, which we can submit for consideration to be added to the design system if it tests well. I think this option will require some review steps w/ the collab cycle since we'd be going "off-brand" and coming up with something custom. We could test it when we do research for any of the upcoming profile work.

This doesn't feel like the best use of our time at this time.

> don't spend much design time on this and replace all edit/cancel buttons styled as links to the secondary button style already in the design system. I believe this was explored in earlier iterations of the design, and the team decided against it for edit buttons because it was overwhelming on the page. I think the phrase "wall of buttons" came up when I asked about this in the past.

Re: "Wall of buttons" — this begs this question for me: isn't that going to be a problem no matter the design if we change this from a link to a button? I'm not trying to be difficult, I'm trying to understand how we can solve this problem with a button if we already tried what we have and people felt that it was too overwhelming in the design. It kind of seems like our only solution is to use the secondary button style whether we prefer that from a visual perspective or not. Thoughts?

**Comment by andaleliz** [comment]
@Samara-Strauss based on the convo yesterday to go ahead with the most simple solution, I went ahead and mocked up the personal and contact info section (has the most buttons) to reflect the secondary button style.  There are a lot of buttons, but it's a worthwhile trade for improved a11y IMO.   

What do you think? I want to be sure we're on the same page before we have the devs implement the fix. We can always improve upon this at some point when we have an official design system component to use that we don't need to design/build ourselves.

![profile - desktop edit button update](https://user-images.githubusercontent.com/55992116/112511964-4abe4c80-8d69-11eb-8f4e-aa7b03845a0d.png)
![profile - mobile edit button update](https://user-images.githubusercontent.com/55992116/112511975-4e51d380-8d69-11eb-92e7-98a257e3695b.jpg)

**Comment by Samara-Strauss** [comment]
Can you get input from our team and the design team, and report back with what people think? It does feel really heavy to me, but it also sounds like we don't have a choice.

**Comment by andaleliz** [comment]
@Samara-Strauss, yep - will report back.

**Comment by andaleliz** [comment]
@Samara-Strauss, spoke with Josh yesterday and got input from the other VSA UX folks as well.  Bottom line is that we need to update the button style from a link to a button - there's just no way around the fact that the page has cognitive challenges that need to be corrected. We're a couple of weeks overdue from our 1 month window to correct level 3 defects as well.

I recommend we move forward with the buttons as mocked-up above for now - it's a bandaid until VSP introduces more lightweight buttons in the design system.

I'm going to paste these screenshots in VSP's ticket as evidence of this problem with a friendly nudge to prioritize this so we can deliver a better experience - many parts of the site will benefit from a new lightweight button.

I also sought input from the other VSA UX designers and they agree this is the best way forward (they also don't dislike the buttons as mocked-up, if that is any consolation for the team. I know the team prefers not to go this route.)

**Comment by Samara-Strauss** [comment]
OK, sounds good. Thanks for letting me know!

**Comment by andaleliz** [comment]
@erikphansen and @sanlouise, I think this is ready for a dev - though not sure who to assign it to. 

We basically need to replace all instances of a button styled like a link with the secondary button style from the design system. This applies to all edit buttons, as well as any cancel buttons floating around.  Do you need me to find instances of the cancel buttons, or is that fairly easy to do programmatically?

**Comment by Samara-Strauss** [comment]
I know @sanlouise was a little light this sprint, so she might be able to take it on. If not, we can move FE work on this to next sprint since we had not considered this in our sprint planning.

**Comment by Samara-Strauss** [comment]
@ajakabcin for awareness — this ticket has been removed from Sprint 44 (I also removed from the epic) since it doesn't require any more design work. Moved to next sprint for FE implementation.

**Comment by andaleliz** [comment]
@Samara-Strauss and @sanlouise there is a related problem on Profile that doesn't have an open issue yet, but I'm sure will get called out in the upcoming a11y audit - the styling of the `Please add your [missing contact information]` buttons has the same problem of material dishonesty.

Here are the current buttons (you can see one with and one without the hover state):
![image.png](https://images.zenhubusercontent.com/5d9def68b6de6d00018d50f2/c969f969-425a-4f71-99b0-3dcacf5f436c)

Here's are a couple of options we can do instead to follow the rest of profile (I don't think it makes sense to put `Please add your [missing contact info]` inside of a secondary style button, that would be gigantic in some cases.

![image.png](https://images.zenhubusercontent.com/5d9def68b6de6d00018d50f2/f96e08c5-1c8d-4f82-93a5-75990b785043)

![image.png](https://images.zenhubusercontent.com/5d9def68b6de6d00018d50f2/25dab249-72b6-4d56-b0dc-42c63ccea102)

Wanted to bring this up now in case it's easier to just go in and fix that while you're making the update to the `edit` buttons.  We'd have one less a11y audit finding to address later and the site would be a little more accessible sooner  :tada:

**Comment by Samara-Strauss** [comment]
@andaleliz thanks for bringing this up! Here are my thoughts:

- I think the general pattern of copy explaining we don't have information/button to update works well in terms of clarity and consistency with the current profile patterns.
- Personally, I think keeping all the buttons as "Edit" makes more sense given the wall of buttons. As for the "blank state" copy explaining we don't have a specific piece of info, let us know what the content folks recommend.

Great work!!

**Comment by andaleliz** [comment]
@Samara-Strauss should we address that with this work too, or create a separate issue to specifically address the "blank state" copy-as-buttons problem? I can add an updated mock-up for that here if needed.


Also, want to confirm with @erikphansen that it's clear to all that we need to update these edit and cancel link-style buttons all throughout profile, even though my mock-up only shows personal and contact info.  I didn't think it'd be necessary to mock-up every single instance, but please let me know if I'm wrong about that.

**Comment by Samara-Strauss** [comment]
@Samara-Strauss should we address that with this work too, or create a separate issue to specifically address the "blank state" copy-as-buttons problem? I can add an updated mock-up for that here if needed.

Will you work with @ajakabcin to create a separate issue for that? Since there are blank states for each issue, you'll need updated copy for all of them. It might get to be too much to track here.

**Comment by ajakabcin** [comment]
> @Samara-Strauss should we address that with this work too, or create a separate issue to specifically address the "blank state" copy-as-buttons problem? I can add an updated mock-up for that here if needed.
> 
> Will you work with @ajakabcin to create a separate issue for that? Since there are blank states for each issue, you'll need updated copy for all of them. It might get to be too much to track here.

Create #22742 for sprint 46 to address this

**Comment by joshkimux** [comment]
<img width="1165" alt="Screen Shot 2021-04-13 at 3 30 32 PM" src="https://user-images.githubusercontent.com/14154792/114609901-510b5e80-9c6d-11eb-9c77-b43f6dc03673.png">

This will also apply to the remove button 🙇

**Comment by Samara-Strauss** [comment]
How do we solve this? Having three buttons seems like a lot.

**Comment by andaleliz** [comment]
I agree, that's a lot of buttons.

@Samara-Strauss or @tressaellen any background you can share about the choice to use a button vs just let people save the empty field?

This is another great example that would benefit from some lightweight button options in our design system @CrystabelReiter  😁

**Comment by andaleliz** [comment]
Also I think this might need to get kicked back to me - I'm not sure how to solve it yet but that's definitely a design (not FE) task.

**Comment by Samara-Strauss** [comment]
> @Samara-Strauss or @tressaellen any background you can share about the choice to use a button vs just let people save the empty field?

I don't know that this had any more thinking behind it beyond this pattern feeling intuitive. Once I've added something, my mind assumes I can delete it, but not necessarily that I would delete it in a field and then save an empty form. That feels weird to me.

> Also I think this might need to get kicked back to me - I'm not sure how to solve it yet but that's definitely a design (not FE) task.

I've removed Sandra as an assignee so you can think through additional use cases, @andaleliz.

**Comment by andaleliz** [comment]
>I've removed Sandra as an assignee so you can think through additional use cases, @andaleliz.
Sounds good @Samara-Strauss. Do you want this to take priority over the other work I have assigned to me in this sprint? (error states + FE docs for facilities and notifications)

**Comment by Samara-Strauss** [comment]
> Sounds good @Samara-Strauss. Do you want this to take priority over the other work I have assigned to me in this sprint (error states + FE docs for facilities and notifications) , or punt this issue for sprint 46?

Punt to sprint 46. Project work should take priority.

**Comment by joshkimux** [comment]
Updated this to include add buttons as well 😅 
<img width="1424" alt="Screen Shot 2021-04-13 at 4 48 01 PM" src="https://user-images.githubusercontent.com/14154792/114619028-1529c680-9c78-11eb-91dd-2c1d0c6b0b5d.png">

**Comment by andaleliz** [comment]
@joshkimux we created #22742 to address the "Add" button - there's a lot of copy involved and it'll likely be too much to track in this issue.  

Based on previous conversations with Josh, I think we're going end up with a lot of big buttons using the design system style, or we need to invest time in designing a new style for secondary buttons in profile.  I'll keep noodling on this for now and dive in head first in sprint 46.

**Comment by Samara-Strauss** [comment]
> Based on previous conversations with Josh, I think we're going end up with a lot of big buttons using the design system style, or we need to invest time in designing a new style for secondary buttons in profile. I'll keep noodling on this for now and dive in head first in sprint 46.

Sounds good to push this to next sprint. We might need to take this one problem at a time — so first tackle the edit buttons (easy), then think about adding new information, then think about the delete state. I don't want the profile to look sloppy and be buttons on buttons on buttons that don't communicate hierarchy re: primary vs. secondary actions, so this might involve some larger pattern changes. Let's take it iteratively and not feel pressure to do it all at once.

Also, should we close this or turn it into an epic in favor of smaller tickets? Like Liz said above, there's a lot to track in this issue. I think it's become too cumbersome already with all these instances of things where links should be buttons and how that might fundamentally change the design of a lot of things in the profile. If we want to break this down further, @andaleliz please work with @ajakabcin to do so!

**Comment by ajakabcin** [comment]
Adding the needs grooming label to flag for further discussion/breaking down

**Comment by ajakabcin** [comment]
@andaleliz and I will be meeting early in the sprint (46) to break out the issues identified in this ticket into more discreet issues to tackle.

**Comment by ajakabcin** [comment]
@joshkimux thank you for all the feedback here! @andaleliz and I have created separate tickets for the issues identified here. I'll close this out and we can track progress in those separately:

1. [FE] Profile: Correct “edit” buttons so they are styled as buttons #23660 
2. [FE] Profile - Align cancel button styling throughout profile so that all buttons look like buttons #23661 
3. [Design] Profile: improve UX for “please add your [missing contact information]” linked text  - #22742 
4. [Design] Profile - Improve UX for “remove” [contact information] button  #23665 
5. [Design] Profile - Determine how to handle profile’s “remove contact info” feature when user is subscribed to notifications #23666

cc @Samara-Strauss

**Comment by Samara-Strauss** [comment]
Thank you, @ajakabcin and @andaleliz! We can prioritize these in sprints as time allows.
